### ISSUE TITLE
Remove database files present in the version control system

### ISSUE THREAD

**Issue by clarete** [opencollective/opencollective#1048]
## Intro
Although it helps a lot having a database with values setup for some tests, it creates some problems that, in my opinion, overweight the advantages. Here's a non exhaustive list of problems I've ran into:

  1. Impossible to know if the test makes sense or not by reading the test itself. You need a SQL client open at the same time with multiple tables open.

  2. We get random errors with queries leaking from other tests. postgres screams when we try to delete a database that is being accessed. I debugged it for a while but never really got to the bottom of it. This is one of the reasons that I can't get the tests running locally (to be stable);

  3. Our database related scripts in bash aren't aware of our configuration thus can't be used within docker and increases the burden on automation in general.  I've made progress on this by adding replacements of the bash scripts written in JavaScript. However, tailoring a script that destroys a database and recreate it in JavaScript is just reimplement the problem.

  4. There are incompatibilities between pg_dump/pg_restore and some versions of the server. Which cause the `incompatible header version` error message once in a while.

## What to do instead

We could create a library that provides helpers to create multiple objects for common cases. [I worked on a draft of it](https://github.com/opencollective/opencollective-api/blob/e02b89fdc88e1d0c196e407e5b63d5c1653c1f43/test/lib/stores.js)

## What needs to be changed
There are some tests that would require being rewritten before we get rid of those files. Here's the list of files:
 * [X] test/graphql.expenses.test.js (21 tests)
 * [x] test/graphql.search.test.js (2 tests)
 * [x] test/graphql.createOrder.test.js (12 tests -- and the nock setup of this file is also very hard to maintain)
 * [x] test/graphql.transaction.test.js (5 tests)
 * [x] test/graphql.collective.test.js (13 tests)
 * [x] test/lib.transactions.test.js (1 test)
 * [x] test/lib.host.test.js (5 tests)
 * [x] test/graphql.invoices.test.js (3 tests)

**Comment by xdamman** [comment]
To give some context: 
The idea was to seed the database with some data. So that when you install Open Collective on your machine, you already have a few collectives to play with with a few individual backers and organizations.

It also makes it easier to test complex methods such as computing the balance of a host across multiple collectives (which would require creating a lot of rows just to set the context up). It's a lazy way to do it and probably not the best.

**Comment by xdamman** [comment]
tldr; I'd keep one `.pgsql` file to seed the database but we shouldn't probably rely on it for the tests.

**Comment by clarete** [comment]
> tldr; I'd keep one .pgsql file to seed the database but we shouldn't probably rely on it for the tests.

Definitely a good idea! I'll keep it there for sure!

**Comment by clarete** [comment]
Alright. I removed all the tests that depended on the `loadDB` utility!
